### ISSUE TITLE
panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by ladung** [karmada-io/karmada#2444]
**What happened**:

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Karmada version: 1.2.2
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`):
- Others:

**Comment by ladung** [comment]
When I run command `local-up-karmada.sh` , it have problem:

```
Can't load /root/.rnd into RNG
140681967710656:error:2406F079:random number generator:RAND_load_file:Cannot open file:../crypto/rand/randfile.c:88:Filename=/root/.rnd
Generating a RSA private key
........+++++
....................................+++++
writing new private key to '/root/.karmada/server-ca.key'
-----
Can't load /root/.rnd into RNG
140354258612672:error:2406F079:random number generator:RAND_load_file:Cannot open file:../crypto/rand/randfile.c:88:Filename=/root/.rnd
Generating a RSA private key
..+++++
...+++++
writing new private key to '/root/.karmada/front-proxy-ca.key'
-----
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x8bfa86]

goroutine 1 [running]:
debug/elf.(*Section).ReadAt(0xc0000b2320?, {0xc00036ac00?, 0x11552e8?, 0x30?}, 0x40?)
	<autogenerated>:1 +0x26
archive/zip.readDirectoryEnd({0xbe5940, 0xc0000fd780}, 0x8a0)
	/usr/local/go/src/archive/zip/reader.go:526 +0xf5
archive/zip.(*Reader).init(0xc000235570, {0xbe5940?, 0xc0000fd780}, 0x8a0)
	/usr/local/go/src/archive/zip/reader.go:97 +0x5c
archive/zip.NewReader({0xbe5940, 0xc0000fd780}, 0x8a0)
	/usr/local/go/src/archive/zip/reader.go:90 +0x5e
github.com/daaku/go%2ezipexe.zipExeReaderElf({0xbe66c0?, 0xc0000140e0}, 0x1155ee8)
	/root/go/pkg/mod/github.com/daaku/go.zipexe@v1.0.0/zipexe.go:128 +0x8b
github.com/daaku/go%2ezipexe.NewReader({0xbe66c0, 0xc0000140e0}, 0x0?)
	/root/go/pkg/mod/github.com/daaku/go.zipexe@v1.0.0/zipexe.go:48 +0x98
github.com/daaku/go%2ezipexe.OpenCloser({0xc000038300?, 0xc000145be0?})
	/root/go/pkg/mod/github.com/daaku/go.zipexe@v1.0.0/zipexe.go:30 +0x57
github.com/GeertJohan/go%2erice.init.0()
	/root/go/pkg/mod/github.com/!geert!johan/go.rice@v1.0.0/appended.go:41 +0x65
Failed to parse input: unexpected end of JSON input
```

**Comment by RainbowMango** [comment]
Hi @ladung , I guess you are using Go 1.19(if not the case, please let me know).

There is an issue addressed at [here](https://github.com/cloudflare/cfssl/issues/1233). 

Could you please run this script with `Go 1.18`?

By the way, we planned to update the default cfssl version to v1.6.2 by this week. After that, you can use Go1.19.

**Comment by ladung** [comment]
> Hi @ladung , I guess you are using Go 1.19(if not the case, please let me know).
> 
> There is an issue addressed at [here](https://github.com/cloudflare/cfssl/issues/1233).
> 
> Could you please run this script with `Go 1.18`?
> 
> By the way, we planned to update the default cfssl version to v1.6.2 by this week. After that, you can use Go1.19.

Hi @RainbowMango , I am using Go 1.17.2 `go version go1.17.12 linux/amd64` and it still trouble

**Comment by RainbowMango** [comment]
I'm not sure if Go1.17 works since our CI job are all running with Go1.18 now.
Can you try again with Go1.18?

By the way, what's the OS version you are using?

**Comment by ladung** [comment]
> I'm not sure if Go1.17 works since our CI job are all running with Go1.18 now. Can you try again with Go1.18?
> 
> By the way, what's the OS version you are using?

My OS is ubuntu 18.04.6, I used Go 1.18.5 but probblem not resolve!
```
-----
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x8bfa86]

goroutine 1 [running]:
debug/elf.(*Section).ReadAt(0xc0003e8000?, {0xc0000fe400?, 0x70?, 0x7f78b10ebfff?}, 0x7f78b0ef1a00?)
	<autogenerated>:1 +0x26
archive/zip.readDirectoryEnd({0xbe5940, 0xc0000a7280}, 0x8a0)
	/usr/local/go/src/archive/zip/reader.go:526 +0xf5
archive/zip.(*Reader).init(0xc0000b2e70, {0xbe5940?, 0xc0000a7280}, 0x8a0)
	/usr/local/go/src/archive/zip/reader.go:97 +0x5c
archive/zip.NewReader({0xbe5940, 0xc0000a7280}, 0x8a0)
	/usr/local/go/src/archive/zip/reader.go:90 +0x5e
github.com/daaku/go%2ezipexe.zipExeReaderElf({0xbe66c0?, 0xc0001280d0}, 0x1155ee8)
	/root/go/pkg/mod/github.com/daaku/go.zipexe@v1.0.0/zipexe.go:128 +0x8b
github.com/daaku/go%2ezipexe.NewReader({0xbe66c0, 0xc0001280d0}, 0x0?)
	/root/go/pkg/mod/github.com/daaku/go.zipexe@v1.0.0/zipexe.go:48 +0x98
github.com/daaku/go%2ezipexe.OpenCloser({0xc0001521b0?, 0xc0001dfbe0?})
	/root/go/pkg/mod/github.com/daaku/go.zipexe@v1.0.0/zipexe.go:30 +0x57
github.com/GeertJohan/go%2erice.init.0()
	/root/go/pkg/mod/github.com/!geert!johan/go.rice@v1.0.0/appended.go:41 +0x65
Failed to parse input: unexpected end of JSON input
root@dung:~/karmada# go version
go version go1.18.5 linux/amd64
```

**Comment by RainbowMango** [comment]
What's the cfssl version on your machine? 
You can get it by command:
```
cfssl version
```

**Comment by ladung** [comment]
> What's the cfssl version on your machine? You can get it by command:
> 
> ```
> cfssl version
> ```
Oh, I was not install `cfssl` before and it had this trouble. So when i installed `cfssl` by command `apt install golang-cfssl`, that work for me! 
Thank you @RainbowMango  for supporting for me!
But i have other problem, ```Unable to connect to the server: net/http: request canceled (Client.Timeout exceeded while awaiting headers)```. What does this mean?

**Comment by RainbowMango** [comment]
> But i have other problem, Unable to connect to the server: net/http: request canceled (Client.Timeout exceeded while awaiting headers). What does this mean?

I didn't meet this issue before. Please show more logs before and after this error, let me see what I can help with.

**Comment by ladung** [comment]
> > But i have other problem, Unable to connect to the server: net/http: request canceled (Client.Timeout exceeded while awaiting headers). What does this mean?
> 
> I didn't meet this issue before. Please show more logs before and after this error, let me see what I can help with.

@RainbowMango it's here
```
Context "kind-karmada-host" renamed to "karmada-host".
Cluster "kind-karmada-host" set.
Image: "swr.ap-southeast-1.myhuaweicloud.com/karmada/karmada-controller-manager:latest" with ID "sha256:b0e86d8848d01bca02bf3a1bc752332f7fdc45a0860e7890763bfdceadfd89ed" not yet present on node "karmada-host-control-plane", loading...
Image: "swr.ap-southeast-1.myhuaweicloud.com/karmada/karmada-scheduler:latest" with ID "sha256:45beed099b1dda6095bff609a94cf751438410426ded57361857ea7bb252d959" not yet present on node "karmada-host-control-plane", loading...
Image: "swr.ap-southeast-1.myhuaweicloud.com/karmada/karmada-descheduler:latest" with ID "sha256:3b26beac5fd2d5f4f82fc1d53f170edfd9dbbdbd739d54fba79b60678b029a8d" not yet present on node "karmada-host-control-plane", loading...
Image: "swr.ap-southeast-1.myhuaweicloud.com/karmada/karmada-webhook:latest" with ID "sha256:914421d0f862043dff1c27d651f0a100b22294997ee582e60f36ab4ade1b7a33" not yet present on node "karmada-host-control-plane", loading...
Image: "swr.ap-southeast-1.myhuaweicloud.com/karmada/karmada-scheduler-estimator:latest" with ID "sha256:d6ee58abf25bee8ef84264a4bf509894ff065def35c7aeb00cb4485b2f5ed632" not yet present on node "karmada-host-control-plane", loading...
Image: "swr.ap-southeast-1.myhuaweicloud.com/karmada/karmada-aggregated-apiserver:latest" with ID "sha256:3077c88a7a578580c7e2a5a589a89d07e8b12c844b9affddf9f1fe369dad671b" not yet present on node "karmada-host-control-plane", loading...
Image: "swr.ap-southeast-1.myhuaweicloud.com/karmada/karmada-search:latest" with ID "sha256:ae6e613ff86b82b893b961201fb890b3e9d1dfa6655f232adddd6768875cec5a" not yet present on node "karmada-host-control-plane", loading...
Can't load /root/.rnd into RNG
140422723330496:error:2406F079:random number generator:RAND_load_file:Cannot open file:../crypto/rand/randfile.c:88:Filename=/root/.rnd
Generating a RSA private key
......................................................+++++
.........+++++
writing new private key to '/root/.karmada/server-ca.key'
-----
Can't load /root/.rnd into RNG
140708050497984:error:2406F079:random number generator:RAND_load_file:Cannot open file:../crypto/rand/randfile.c:88:Filename=/root/.rnd
Generating a RSA private key
..................................................+++++
................................................+++++
writing new private key to '/root/.karmada/front-proxy-ca.key'
-----
2022/08/30 10:59:09 [INFO] generate received request
2022/08/30 10:59:09 [INFO] received CSR
2022/08/30 10:59:09 [INFO] generating key: rsa-2048
2022/08/30 10:59:10 [INFO] encoded CSR
2022/08/30 10:59:10 [INFO] signed certificate with serial number 66449952171498119416375551999164535391239627129
2022/08/30 10:59:10 [INFO] generate received request
2022/08/30 10:59:10 [INFO] received CSR
2022/08/30 10:59:10 [INFO] generating key: rsa-2048
2022/08/30 10:59:10 [INFO] encoded CSR
2022/08/30 10:59:10 [INFO] signed certificate with serial number 640972549622150417416887743020628774930878933306
namespace/karmada-system created
serviceaccount/karmada-controller-manager created
serviceaccount/karmada-scheduler created
serviceaccount/karmada-webhook created
clusterrole.rbac.authorization.k8s.io/karmada-controller-manager created
clusterrolebinding.rbac.authorization.k8s.io/karmada-controller-manager created
secret/karmada-cert-secret created
secret/kubeconfig created
secret/webhook-cert created
statefulset.apps/etcd created
service/etcd-client created
service/etcd created
wait the etcd ready...
pod/etcd-0 condition met

Apply dynamic rendered apiserver service in /tmp/tmp.TVNvakQtmO/karmada-apiserver.yaml.
deployment.apps/karmada-apiserver created
service/karmada-apiserver created
wait the karmada-apiserver ready...
pod/karmada-apiserver-5666bb88fd-6qzcc condition met

Karmada API Server's IP is: 172.18.0.2, host cluster type is: local
Cluster "karmada-apiserver" set.
User "karmada-apiserver" set.
Context "karmada-apiserver" created.
deployment.apps/karmada-kube-controller-manager created
deployment.apps/karmada-aggregated-apiserver created
service/karmada-aggregated-apiserver created
wait the karmada-aggregated-apiserver ready...
pod/karmada-aggregated-apiserver-75c95c5fb-ggtks condition met
pod/karmada-aggregated-apiserver-75c95c5fb-kbwvq condition met
deployment.apps/karmada-search created
service/karmada-search created
wait the karmada-search ready...
pod/karmada-search-6969db88c5-557bj condition met
pod/karmada-search-6969db88c5-lzrwq condition met
namespace/karmada-system created
customresourcedefinition.apiextensions.k8s.io/clusteroverridepolicies.policy.karmada.io created
customresourcedefinition.apiextensions.k8s.io/clusterpropagationpolicies.policy.karmada.io created
customresourcedefinition.apiextensions.k8s.io/clusterresourcebindings.work.karmada.io created
customresourcedefinition.apiextensions.k8s.io/federatedresourcequotas.policy.karmada.io created
customresourcedefinition.apiextensions.k8s.io/multiclusteringresses.networking.karmada.io created
customresourcedefinition.apiextensions.k8s.io/overridepolicies.policy.karmada.io created
customresourcedefinition.apiextensions.k8s.io/propagationpolicies.policy.karmada.io created
customresourcedefinition.apiextensions.k8s.io/resourcebindings.work.karmada.io created
customresourcedefinition.apiextensions.k8s.io/resourceinterpreterwebhookconfigurations.config.karmada.io created
customresourcedefinition.apiextensions.k8s.io/serviceexports.multicluster.x-k8s.io created
customresourcedefinition.apiextensions.k8s.io/serviceimports.multicluster.x-k8s.io created
customresourcedefinition.apiextensions.k8s.io/works.work.karmada.io created
mutatingwebhookconfiguration.admissionregistration.k8s.io/mutating-config created
validatingwebhookconfiguration.admissionregistration.k8s.io/validating-config created
Unable to connect to the server: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
root@dung:~/karmada# docker ps
```

**Comment by elinx** [comment]
Meet the same problem here `Unable to connect to the server: net/http: request canceled (Client.Timeout exceeded while awaiting headers)`, it happens because I  use go1.19.2 at the first run and re-install cfssl(build with go1.19.2) after I saw this issue but it still exists. Then I downgraded using go1.18(with cfssl built using go1.18), and it solved.

**Comment by RainbowMango** [comment]
@elinx I guess the issue you met probably has been fixed by #2462.

@ladung could you please try again with the latest code? Before retrying, you can remove the previous `cfssl`(to find where it is, run `command -v cfssl`), and `hack/local-up-karmada.sh` can re-install it.

**Comment by XiShanYongYe-Chang** [comment]
This issue has been inactive for a long time. Close it first. If you still encounter problems, please feel free to reopen it.
/close
